pkg/gitops/audit_handler.go

```diff
@@ -14,6 +14,82 @@
 
 package gitops
 
+import (
+	"fmt"
+	"bytes"
+    "errors"
+	"encoding/json"
+	"k8s.io/klog/v2"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+)
+
 func (cr *CustomRepo) HandleEventList(jsonstring []byte) error {
+	cr.Mutex.Lock()
+	defer cr.Mutex.Unlock()
+	eventList := auditv1.EventList{}
+	jsonstring = bytes.TrimPrefix(jsonstring, []byte("\xef\xbb\xbf"))
+	err := json.Unmarshal(jsonstring, &eventList)
+	if err != nil {
+		klog.ErrorS(err, "unable to unmarshal json into event list struct")
+		return err
+	}
+	for _, event := range eventList.Items {
+		if event.Stage != "ResponseComplete" ||
+			event.ResponseStatus.Status == "Failure" ||
+			event.User.Username == cr.ServiceAccount {
+			klog.V(2).InfoS("Audit event skipped (audit Stage != ResponseComplete, audit ResponseStatus != Success, or audit produced by rollback)")
+			continue
+		}
+		if cr.RollbackMode {
+			klog.V(2).ErrorS(err, "could not handle event with rollback in progress")
+			return fmt.Errorf("rollback in progress")
+		}
+        if err = cr.HandleEvent(event); err != nil {
+			klog.V(2).ErrorS(err, "could not handle event")
+            return err
+        }
+	}
 	return nil
 }
+
+func (cr *CustomRepo) HandleEvent(event auditv1.Event) (error) {
+    user := event.User.Username
+    email := event.User.Username + "+" + event.User.UID + "@audit.antrea.io"
+	message := resourceMap[event.ObjectRef.Resource+event.ObjectRef.APIGroup] + event.ObjectRef.Namespace + "/" + event.ObjectRef.Name
```

Comment:
there is inconsistent formatting. Usually you can ask your editor to run `go fmt` automatically when you save a file. Running `make golangci-fix` should take care of this as well (notice that the golangci CI check is failing)